src/users/index.ts

```diff
@@ -18,36 +46,139 @@ export interface Users {
+    delete: (userId: string) =>
+      connection
+        .delete(`/users/db/${userId}`, null)
+        .then(() => true)
+        .catch(() => false),
+    rotateKey: (userId: string) =>
+      connection.postNoBody<APIKeyResponse>(`/users/db/${userId}/rotate-key`).then((resp) => resp.apikey),
+    activate: (userId: string) =>
+      connection
+        .postNoBody(`/users/db/${userId}/activate`)
+        .then(() => true)
+        .catch(allowCode(409)),
+    deactivate: (userId: string) =>
+      connection
+        .postNoBody(`/users/db/${userId}/deactivate`)
+        .then(() => true)
+        .catch(allowCode(409)),
+    byName: (userId: string) => connection.get<WeaviateDBUser>(`/users/db/${userId}`, true).then(Map.dbUser),
+    listAll: () => connection.get<WeaviateDBUser[]>('/users/db', true).then(Map.dbUsers),
+  };
+};
+
+const oidc = (connection: ConnectionREST): OIDCUsers => {
+  const ns = namespacedUsers(connection);
+  return {
+    getAssignedRoles: (userId: string, includePermissions?: boolean) =>
+      ns.getAssignedRoles('oidc', userId, includePermissions),
+    assignRoles: (roleNames: string | string[], userId: string) => ns.assignRoles(roleNames, userId, 'oidc'),
+    revokeRoles: (roleNames: string | string[], userId: string) => ns.revokeRoles(roleNames, userId, 'oidc'),
+  };
+};
+
+/** Internal interface for operations that MAY accept a 'db'/'oidc' namespace. */
+interface NamespacedUsers {
+  getAssignedRoles: (
+    userType: UserTypeInternal,
+    userId: string,
+    includePermissions?: boolean
+  ) => Promise<Record<string, Role>>;
+  assignRoles: (roleNames: string | string[], userId: string, userType?: UserTypeInternal) => Promise<void>;
+  revokeRoles: (roleNames: string | string[], userId: string, userType?: UserTypeInternal) => Promise<void>;
+}
+
+const baseUsers = (connection: ConnectionREST): UsersBase => {
+  const ns = namespacedUsers(connection);
+  return {
+    assignRoles: (roleNames: string | string[], userId: string) => ns.assignRoles(roleNames, userId),
+    revokeRoles: (roleNames: string | string[], userId: string) => ns.revokeRoles(roleNames, userId),
+  };
+};
+
+const namespacedUsers = (connection: ConnectionREST): NamespacedUsers => {
+  return {
+    getAssignedRoles: (userType: UserTypeInternal, userId: string, includePermissions?: boolean) =>
+      connection
+        .get<WeaviateRole[]>(
+          `/authz/users/${userId}/roles/${userType}${includePermissions ? '?&includeFullRoles=true' : ''}`
+        )
+        .then(Map.roles),
+    assignRoles: (roleNames: string | string[], userId: string, userType?: UserTypeInternal) =>
       connection.postEmpty(`/authz/users/${userId}/assign`, {
         roles: Array.isArray(roleNames) ? roleNames : [roleNames],
+        userType: userType,
       }),
-    revokeRoles: (roleNames: string | string[], userId: string) =>
+    revokeRoles: (roleNames: string | string[], userId: string, userType?: UserTypeInternal) =>
```

Comment:
wdyt about having an optional object type here for `userType`? Is it as useful as the other instance?